src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,173 @@
+            user.setLogin(login);
+            user.setPassword("validPassword");
+            user.setAge(20);
+
+            RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                    registrationService.register(user));
+
+            assertEquals("Login length must be at least 6 characters.", exception.getMessage());
+        }
+    }
+
+    @Test
+    void registerLoginExists_NotOk() throws RegistrationException {
+        User existingUser = new User();
+        existingUser.setLogin("existingUser");
+        existingUser.setPassword("password123");
+        existingUser.setAge(20);
+        registrationService.register(existingUser);
+
+        User newUserWithExistingLogin = new User();
+        newUserWithExistingLogin.setLogin("existingUser");
+        newUserWithExistingLogin.setPassword("newPassword123");
+        newUserWithExistingLogin.setAge(25);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () -> {
+            registrationService.register(newUserWithExistingLogin);
+        });
+
+        assertEquals("Login already exists.", exception.getMessage());
+    }
+
+
+    @Test
+    void registerPassword_NotOk() throws RegistrationException {
+        String[] invalidPasswords = {"1", "ab", "son", "user", "hell0"};
+
+        for (String password : invalidPasswords) {
+            user.setLogin("validLog");
+            user.setPassword(password);
+            user.setAge(20);
+
+            RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                    registrationService.register(user));
+
+            assertEquals("Password length must be at least 6 characters.", exception.getMessage());
+        }
+    }
+
+    @Test
+    void registerUserAge_NotOk() {
+        for (int i = 0; i < 18; i++) {
+            user.setLogin("validLogin");
+            user.setPassword("validPassword");
+            user.setAge(i);
+
+            RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                    registrationService.register(user));
+
+            assertEquals("Age must be at least 18 years old.", exception.getMessage());
+        }
+
+        int validAge = 18;
+        user.setLogin("validLogin");
+        user.setPassword("validPassword");
+        user.setAge(validAge);
+
+        assertDoesNotThrow(() -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_Ok() {
+        user.setLogin("validUser");
+        user.setPassword("validPassword123");
+        user.setAge(20);
+
+        User registeredUser = assertDoesNotThrow(() -> registrationService.register(user));
+
+        assertNotNull(registeredUser);
+        assertEquals("validUser", registeredUser.getLogin());
+        assertEquals("validPassword123", registeredUser.getPassword());
+        assertEquals(20, registeredUser.getAge());
```

Comment:
The test `register_Ok` is good for checking a successful registration. However, ensure that after successful registration, the user was actually added to storage and the `register` method returned the correct user. This can be verified by checking if the user is retrievable from the storage after registration.